atest/testdata/running/for.robot

```diff
@@ -317,6 +324,128 @@ For In Range With Expressions Containing Floats
     \    @{result} =    Create List    @{result}    ${i}
     Should Be True    @{result} == [3.14, 1.14, -0.86]
 
+For In Zip
+    @{items}=    Create List    a    b    c    d
+    @{things}=    Create List    e    f    g    h
+    : FOR    ${item}    ${thing}    IN ZIP    ${items}    ${things}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}
+    Should Be True    @{result} == ['a:e', 'b:f', 'c:g', 'd:h']
+
+For In Zip With Uneven Lists
+    [Documentation]    This will ignore any elements after the shortest
+    ...    list ends, just like with Python's zip()
+    @{items}=    Create List    a    b    c
+    @{things}=    Create List    d    e    f    g    h
+    : FOR    ${item}    ${thing}    IN ZIP    ${items}    ${things}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}
+    Should Be True    @{result} == ['a:d', 'b:e', 'c:f']
+
+For In Zip With 3 Lists
+    @{items}=    Create List    a    b    c    d
+    @{things}=    Create List    e    f    g    h
+    @{stuffs}=    Create List    1    2    3    4    5
+    : FOR    ${item}    ${thing}    ${stuff}    IN ZIP    ${items}    ${things}    ${stuffs}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}:${stuff}
+    Should Be True    @{result} == ['a:e:1', 'b:f:2', 'c:g:3', 'd:h:4']
+
+For In Zip With Other Iterables
+    [Documentation]    Handling non-lists. Should accept anything iterable
+    ...    except strings and fail with a clear error message if invalid
+    ...    data given. You can use utils.is_list_like to verify inputs.
+    @{range}=    Evaluate    (i for i in range(10))
+    @{range2}=    Evaluate    (10, 11, 12, 13, 14, 15, 16, 17, 18, 19)
```

Comment:
In the current for this test actually doesn't test that it should. When you assign a list variable like `@{range}`, Robot will actually convert the value to a real list. You should assign these iterables to a scalar variable instead. It would also be a little cleaner to use more explicit variable names like `${generatot}` and `${tuple}` in this context.